### PR TITLE
[IMPROVEMENT] Chat list UI improvements

### LINKED ISSUE #2109: Add background color to ChatRoom separator

### PR DESCRIPTION
**Pull request by eliezer-of**
@RocketChat/android

Improves the chat list type section/separator view.

Closes #2109

### CODE REVIEW

File: app/src/main/res/layout/item_chatroom_header.xml [hunk 1]
```diff
@@ -14,14 +15,14 @@
...
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:padding="8dp"
+        android:padding="16dp"
         android:paddingEnd="@dimen/screen_edge_left_and_right_padding"
         android:paddingStart="@dimen/screen_edge_left_and_right_padding"
         android:text="@string/chatroom_header" />
 
-    <View
+    <!--<View
```
Comment (philipbrito) [inline app/src/main/res/layout/item_chatroom_header.xml]: Isn't it necessary? Then remove this commented code.

**Review by philipbrito — CHANGES_REQUESTED** [review]

**Review by philipbrito — CHANGES_REQUESTED** [review]
There is something to be improved by seeing the screenshot above. Note that a divider is being shown below the channels separator (sandbox channel).

**Comment by philipbrito** [comment]
@eliezer-of Here is a reference on how this screen should looks like.

screen shot 2019-03-08 at 16 12 20

File: app/src/main/res/layout/item_chatroom_header.xml (hunk 1 above)
Comment (eliezer-of) [reply app/src/main/res/layout/item_chatroom_header.xml]: > There is something to be improved by seeing the screenshot above. Note that a divider is being shown below the channels separator (sandbox channel).

@filipedelimabrito Yes, I had seen it too! I was thinking about opening a new issue, but with your suggestion I'll fix it in this PR. What do you think?

Comment (philipbrito) [reply app/src/main/res/layout/item_chatroom_header.xml]: It really needs to be fixed on this PR. Would be grate if you change that screen to what we have here 
https://github.com/RocketChat/Rocket.Chat.Android/pull/2110#issuecomment-470986869

**Comment by eliezer-of** [comment]
@filipedelimabrito  the changes are correct?

**Comment by eliezer-of** [comment]
@filipedelimabrito the changes are correct? :sweat_smile:

**Comment by philipbrito** [comment]
Could you provide a screenshot with your most recent changes @eliezer-of ? So I can see how it looks like.

**Comment by eliezer-of** [comment]
> Could you provide a screenshot with your most recent changes @eliezer-of ? So I can see how it looks like.

sure :sweat_smile:


Screenshot_1552403418

**Comment by eliezer-of** [comment]
> Could you provide a screenshot with your most recent changes @eliezer-of ? So I can see how it looks like.

@filipedelimabrito sure! :sweat_smile:

Screenshot_1552403418

**Comment by eliezer-of** [comment]
> Could you provide a screenshot with your most recent changes @eliezer-of ? So I can see how it looks like.

@filipedelimabrito sure! :sweat_smile: 

Screenshot_1552403418

File: app/src/main/java/chat/rocket/android/widget/DividerItemDecoration.kt [hunk 2]
```diff
@@ -56,10 +57,13 @@ class DividerItemDecoration() : RecyclerView.ItemDecoration() {
             if (isLastView(child, parent))
                 continue
 
+            if (isViewTypeHeader(child, parent))
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/widget/DividerItemDecoration.kt]: You don't need to add another conditional here. Just define it:

```
if (isLastView(child, parent) || isViewTypeHeader(child, parent)) 
   continue
```

**Review by philipbrito — CHANGES_REQUESTED** [review]
Screen is looking very good! Just need to address one more thing before the merge.

File: app/src/main/java/chat/rocket/android/widget/DividerItemDecoration.kt (hunk 2 above)
Comment (eliezer-of) [reply app/src/main/java/chat/rocket/android/widget/DividerItemDecoration.kt]: @filipedelimabrito  sorry about that :man_facepalming:

**Review by philipbrito — APPROVED** [review]
Thanks for submitting this PR @eliezer-of!

**Pull request merged**